R/worldfootballr_helpers.R

```diff
@@ -271,6 +271,54 @@ tm_team_staff_urls <- function(team_urls, staff_role) {
 }
 
 
+#' Get Understat available teams
```

Comment:
How would this function scale across seasons? Hard coding team names means every season this would need to be re-done?

I don't love this function tbh

Reply:
I understand you. I found an option how to do this without hard coding.

```R
understat_available_teams <- function(leagues){
  correct_leagues <- c("EPL", "La liga", "Bundesliga", "Serie A", "Ligue 1", "RFPL")

  teams_list <- list()

  for(lg in leagues){
    if(!lg %in% correct_leagues){warning(glue::glue("League {lg} not found. Please check understats.com for the correct league name")); next}
    match_url <- switch (lg,
                         'EPL' = 'https://understat.com/team/Arsenal/2023',
                         'La liga' = 'https://understat.com/team/Barcelona/2023',
                         'Bundesliga' = 'https://understat.com/team/Bayern_Munich/2023',
                         'Serie A' = 'https://understat.com/team/AC_Milan/2023',
                         'Ligue 1' = 'https://understat.com/team/Paris_Saint_Germain/2023',
                         'RFPL' = 'https://understat.com/team/Spartak_Moscow/2023'
    )
    team_page <- tryCatch(.load_page(match_url), error = function(e) NA)
    teams_list[[lg]] <- team_page %>% rvest::html_nodes(".header-wrapper") %>% html_text() %>% gsub("([\n\t])|(\\d{4}/\\d{4})", "", .) %>% gsub('(?<!\\s)([[:upper:]])', '(&&)\\1', ., perl = TRUE) %>%
      strsplit(., "\\(&&\\)", perl = TRUE) %>% unlist(.) %>% .[2:length(.)] %>%  unlist(.)
  }
  if(length(leagues) == 1){
    return(unlist(teams_list, use.names = FALSE))
  }

  return(teams_list)
}
```

Reply:
@JaseZiv

Reply:
I've been thinking about this for the last few days... I just think it feels a little out of place in worldfootballR, but happy to keep it in place